File: Algorithms/Python/Taller03/FloodFill.py
class FloodFill():

    def PrintFill(self, Figure, Row, Column, Color):
        self.Print(self.Fill(Figure, Row, Column, Color), 0)
    
    def Fill(self, Figure, Row, Column, Color):
        return self.Floodfill(Figure, Row, Column, Color)
    
    def Floodfill(self, Figure, Row, Column, Color):

        if Row < len(Figure) and Column < len(Figure):
            if Color == Figure[Row][Column]: return Figure
        else: return Figure

        OldColor = Figure[Row][Column]
        Figure[Row][Column] = Color
        
        if Row < len(Figure) - 1:
            if OldColor == Figure[Row+1][Column]:
                self.Floodfill(Figure, Row + 1, Column, Color)
        if Row > 0:
            if OldColor == Figure[Row-1][Column]:
                self.Floodfill(Figure, Row - 1, Column, Color)
        if Column < len(Figure[Row]) - 1:
            if OldColor == Figure[Row][Column+1]:
                self.Floodfill(Figure, Row, Column + 1, Color)
        if Column > 0:
            if OldColor == Figure[Row][Column-1]:
                self.Floodfill(Figure, Row, Column - 1, Color)

        return Figure

    def Print(self, Entry, Swivel):
        if(Swivel < len(Entry)):
            print(Entry[Swivel])
            self.Print(Entry, Swivel + 1)

def main():
    Ff = FloodFill()
    ListMain = []
    print("Write each row of the matrix separated by spaces")
    print("Write False when you've finished of set the values of the matrix")

    while True:
        Lst = input()

        if Lst == "False": break

        ListMain.append(Lst.split())

    for i in range(len(ListMain)):
        for j in range(len(ListMain[i])):
            ListMain[i][j] = int(ListMain[i][j])
    
    Row = int(input("Set the Row you want to change (Starts in 0): "))
    Column = int(input("Set the Column you want to change (Starts in 0): "))
    Color = int(input("Set the color you want to use: "))
    
    Ff.PrintFill(ListMain, Row, Column, Color)

if __name__ == "__main__":
    main()

File: Algorithms/Java/Taller03/FloodFill.java
public class FloodFill
{
    
    /**
     * Print the matrix
     */
    public void PrintFill(int[][] Figure, int Row, int Column, int Color){
        this.Print(this.Floodfill(Figure, Row, Column, Color));
    }
    
    public int[][] Fill(int[][] Figure, int Row, int Column, int Color){
        return this.Floodfill(Figure, Row, Column, Color);
    }
    
    private int[][] Floodfill(int[][] Figure, int Row, int Column, int Color){
        if(Row < Figure.length && Column < Figure[Row].length){ 
            if( Color == Figure[Row][Column])
            return Figure;
        } else return Figure;
        
        int oldColor = Figure[Row][Column];
        Figure[Row][Column] = Color;
        
        if(Row < Figure.length-1){
            if(oldColor == Figure[Row+1][Column])
            Floodfill(Figure, Row + 1, Column, Color);
        }
        if(Column < Figure[Row].length-1){
            if(oldColor == Figure[Row][Column+1])
            Floodfill(Figure, Row, Column + 1, Color);
        }
        if(Row > 0){
            if(oldColor == Figure[Row-1][Column])
            Floodfill(Figure, Row - 1, Column, Color);
        }
        if(Column > 0){
            if(oldColor == Figure[Row][Column-1])
            Floodfill(Figure, Row, Column - 1, Color);
        }
        
        return Figure;
    }
    
    private void Print(int[][] Arr){
        
        for(int z = 0; z < Arr.length; z++){
            for(int i = 0; i < Arr[z].length; i++){
                if(Arr.length == 1)
                System.out.println("[" + Arr[z][i] + "]");            
                else if(i == Arr[z].length - 1)
                System.out.println(Arr[z][i] + "]");
                else if(i == 0)
                System.out.print("[" + Arr[z][i] + ", ");
                else
                System.out.print(Arr[z][i] + ", ");
            }
        }
    }
}
